src/DotNetWorker/Diagnostics/GrpcFunctionsHostLogger.cs

```diff
@@ -2,47 +2,75 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.Threading.Channels;
+using Microsoft.Azure.Functions.Worker.Diagnostics;
 using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
 using Microsoft.Extensions.Logging;
 using static Microsoft.Azure.WebJobs.Script.Grpc.Messages.RpcLog.Types;
 
 namespace Microsoft.Azure.Functions.Worker.Logging
 {
-    public class InvocationLogger : ILogger
+    /// <summary>
+    /// A logger that sends logs back to the Functions host.
+    /// </summary>
+    internal class GrpcFunctionsHostLogger : ILogger
     {
-        private string _invocationId;
-        private ChannelWriter<StreamingMessage> _channelWriter;
+        private readonly string _category;
+        private readonly ChannelWriter<StreamingMessage> _channelWriter;
+        private IExternalScopeProvider _scopeProvider;
 
-        public InvocationLogger(string invocationId, ChannelWriter<StreamingMessage> channelWriter)
+        public GrpcFunctionsHostLogger(string category, ChannelWriter<StreamingMessage> channelWriter, IExternalScopeProvider scopeProvider)
         {
-            _invocationId = invocationId;
-            _channelWriter = channelWriter;
+            _category = category ?? throw new ArgumentNullException(nameof(category));
+            _channelWriter = channelWriter ?? throw new ArgumentNullException(nameof(channelWriter));
+            _scopeProvider = scopeProvider ?? throw new ArgumentNullException(nameof(scopeProvider));
         }
 
         public IDisposable BeginScope<TState>(TState state)
         {
-            return EmptyDisposable.Instance;
+            // The built-in DI wire-up guarantees that scope provider will be set.
+            return _scopeProvider!.Push(state);
         }
 
         public bool IsEnabled(LogLevel logLevel)
         {
-            return true;
+            return logLevel != LogLevel.None;
```

Comment:
We should probably have a todo here as we'll eventually receive the log level from the host.

Reply:
I don't think that will change anything here, though. My idea was to add a Filter that uses that (which happens outside of this call).